File: app/src/main/java/com/orangepenguin/boilerplate/rx/AndroidRxSchedulers.java
package com.orangepenguin.boilerplate.rx;

import rx.Scheduler;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public final class AndroidRxSchedulers implements RxSchedulers {
    @Override
    public Scheduler io() {
        return Schedulers.io();
    }

    @Override
    public Scheduler mainThread() {
        return AndroidSchedulers.mainThread();
    }
}

File: app/src/main/java/com/orangepenguin/boilerplate/rx/AndroidRxSchedulers.kt
package com.orangepenguin.boilerplate.rx

import rx.Scheduler
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

class AndroidRxSchedulers : RxSchedulers {
    override fun io(): Scheduler = Schedulers.io()
    override fun mainThread(): Scheduler = AndroidSchedulers.mainThread()
}
